### PR TITLE
model zoo page

### PR DESCRIPTION
**Pull request by szha**
## Description ##
reorganize website content

### Essentials ###
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] Code is well-documented

### Changes ###
- [x] reorganize main page
- [x] add model zoo page to summarize model offerings

### CODE REVIEW

**Comment by mli** [comment]
Job PR-357/2 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-357/2/index.html

**Comment by mli** [comment]
Job PR-357/4 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-357/4/index.html

File: docs/index.rst [hunk 1]
```diff
@@ -19,11 +19,19 @@ NLP. Otherwise, please refer to an introduction course such as
 
 .. note::
 
-   This project is at an early stage and is under active development.
-   Please expect that it will be updated frequently. Contributions are welcome.
+   We can be found on `Github <https://github.com/dmlc/gluon-nlp>`_. This project is at an early stage
```
Comment (eric-haibin-lin) [inline docs/index.rst]: We should show the list of papers/models we implemented (e.g. https://github.com/pytorch/fairseq#introduction)

File: docs/index.rst [hunk 2]
```diff
@@ -69,38 +77,56 @@ computes the cosine similarity between two words.
...
+
+   examples/word_embedding/word_embedding.ipynb
+   examples/word_embedding/word_embedding_training.ipynb
+   examples/language_model/language_model.ipynb
+   examples/sentiment_analysis/sentiment_analysis.ipynb
+   examples/nmt/gnmt.ipynb
+
+
+.. toctree::
+   :maxdepth: 2
```
Comment (eric-haibin-lin) [inline docs/index.rst]: It's now kind of lengthy for the index page. What about maxdepth =1

File: docs/index.rst [hunk 3]
```diff
@@ -69,38 +77,56 @@ computes the cosine similarity between two words.
...
+   api/notes/data_api.rst
+   api/notes/vocab_emb.rst
+   api/notes/sequence_sampling.rst
+
+
+.. toctree::
+   :maxdepth: 1
+   :caption: Scripts
+
+   scripts/word_embeddings/word_embeddings
```
Comment (eric-haibin-lin) [inline docs/index.rst]: Word Embedding Toolkit -> Word Embedding 
Sentiment Analysis via fine-tuning w/ bucketing -> Sentiment Analysis

Comment (eric-haibin-lin) [reply docs/index.rst]: Sequence Sampling -> Text Generation

**Comment by mli** [comment]
Job PR-357/7 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-357/7/index.html

**Comment by mli** [comment]
Job PR-357/29 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-357/29/index.html

**Comment by eric-haibin-lin** [comment]
Tutorials:
```
Using Pre-trained Word Embeddings
Word Embeddings Training and Evaluation
Language Modeling using GluonNLP
Sentiment Analysis (SA) with pre-trained Language Model (LM)
Google NMT on IWSLT 2015 English-Vietnamese Translation
Transformer based Machine Translation Using GluonNLP

```
-> 

```
Pre-trained Word Embeddings
Word Embeddings Training and Evaluation
Language Modeling 
Sentiment Analysis
Google Neural Machine Translation
Transformer based Machine Translation
```



ModelZoo:
```
Word Embedding
Text Generation
Language Model
Sentiment Analysis
Machine Translation
Text Classification
```
->
```
Word Embedding
Language Model
Sentiment Analysis
Machine Translation
Text Classification
```

API Usage:
```
Data API — A Sentiment Analysis Example
Vocabulary and Embedding
Sentence Generation from Language Model
```
->
```
Data Loading API
Vocabulary and Embedding API
Sentence Generation API
```

**Comment by mli** [comment]
Job PR-357/30 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-357/30/index.html

**Comment by mli** [comment]
Job PR-357/33 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-357/33/index.html

File: docs/index.rst [hunk 4]
```diff
@@ -13,17 +13,34 @@ research ideas and products based on these models. This toolkit offers four main
...
+
+   You can find our the doc for our master development branch `here <http://gluon-nlp.mxnet.io/master/index.html>`_.
+
 
 Installation
-------------------
+------------
 
 GluonNLP relies on the recent version of MXNet. The easiest way to install MXNet
 is through `pip <https://pip.pypa.io/en/stable/installing/>`_. The following
```
Comment (eric-haibin-lin) [inline docs/index.rst]:
```
pip install --pre --upgrade mxnet
```
->
```
mxnet 1.3.0
```

**Comment by szha** [comment]
FYI, @cgraywang @leezu @szhengac @sxjscience there's a bit of re-organizing and renaming in this PR.

File: docs/index.rst (hunk 4 above)
Comment (leezu) [reply docs/index.rst]: The next release may again include code relying on the --pre version, so perhaps leaving it the same would be less confusing?

**Comment by leezu** [comment]
Thanks for your work on this @szha! I noticed that the API Usage Examples (Data Loading API, Vocabulary and Embedding API, Text Sequence Generation API) do not link to the API Documentation for the classes that they mention. (For example `SimpleDataset` at http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-357/33/api/notes/data_api.html ). As those API Usage Examples are now more prominently placed thanks to your work, it may be nice to introduce links?

**Comment by szha** [comment]
@leezu good point. Let me give it a try at linking the classes.

**Comment by mli** [comment]
Job PR-357/34 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-357/34/index.html

**Comment by szha** [comment]
@leezu actually let me get this one in first, as this PR has some CI fixes (such as distributing working among GPUs, reducing test size, etc.)

**Comment by leezu** [comment]
@szha, sounds good, thanks!

**Pull request merged**